### PR TITLE
refactor: inlined xor inplace

### LINKED ISSUE #60: use `buffer-xor/inplace.js` instead of default `buffer-xor`
https://github.com/webtorrent/bittorrent-protocol/blob/13020789bd216529a621d89b63f595bbc5ed4313/index.js#L11
```diff
- const xor = require('buffer-xor') 
+ const xor = require('buffer-xor/inplace.js') 
```

I do not see `hash2Buffer` being used for something else, so it might be a improvement to just modify the original array.
https://github.com/webtorrent/bittorrent-protocol/blob/13020789bd216529a621d89b63f595bbc5ed4313/index.js#L239-L241

---

What i think is better with inplace.js is that it can support plain Uint8Array or any other array for that matter... 
& it don't depend on safe-buffer, but still require you to install it...

### PR DESCRIPTION
**Pull request by jimmywarting**
less dependency on safe-buffer

**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[x] Other, please explain: https://github.com/brave/brave-browser/issues/5490

**What changes did you make? (Give an overview)**
Inlined xor/inplace

**Which issue (if any) does this pull request address?**
closes #60

**Is there anything you'd like reviewers to focus on?**
The first buffer is now mutated (just a.heads up)
(fewer allocation)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,6 +29,11 @@ const VC = Buffer.from([0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00])
 const CRYPTO_PROVIDE = Buffer.from([0x00, 0x00, 0x01, 0x02])
 const CRYPTO_SELECT = Buffer.from([0x00, 0x00, 0x00, 0x02]) // always try to choose RC4 encryption instead of plaintext
 
+function xor (a, b) {
+  for (let i = 0, len = a.length; i < len; i++) a[i] = a[i] ^ b[i]
+  return b
```
Comment (alxhotel) [inline index.js:34]: Shouldn't this be `a` ? And if so, why are the tests passing ?
```suggestion
  return a
```

File: index.js [hunk 2]
```diff
@@ -30,6 +29,11 @@ const VC = Buffer.from([0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00])
 const CRYPTO_PROVIDE = Buffer.from([0x00, 0x00, 0x01, 0x02])
 const CRYPTO_SELECT = Buffer.from([0x00, 0x00, 0x00, 0x02]) // always try to choose RC4 encryption instead of plaintext
 
+function xor (a, b) {
+  for (let i = 0, len = a.length; i < len; i++) a[i] = a[i] ^ b[i]
```
Comment (alxhotel) [inline index.js:33]: Shouldn't we use `len = Math.max(a.length, b.length)`?

File: index.js (hunk 1 above)
Comment (jimmywarting) [reply index.js:34]: hmm, it should be a...

File: index.js (hunk 2 above)
Comment (jimmywarting) [reply index.js:33]: They are always equal length? didn't see it as necessary

File: index.js (hunk 1 above)
Comment (jimmywarting) [reply index.js]: I changed it... Maybe don't have a test for checking the xor:ed hash anywhere?

**Review by DiegoRBaquero — APPROVED** [review]

**Pull request merged**